### ISSUE TITLE
VST3 plain gui: serious problems with a number of VST3 plugins

### ISSUE THREAD

**Issue by Paul-Licameli** [audacity/audacity#2555]
@DavidBailes wrote:

I've had a look at the plain gui, and have found serious problems using a couple of the free plugins from here:
https://www.voxengo.com/group/free-vst-plugin-download/

I downloaded and installed both the vst and vst3 versions of the Mastering Graphic Equalizer, and the Tube Amplifier.
For the VST versions, the existing plain gui was OK.
However for the VST3 versions the plain gui is definitely not OK:
1. Values not in the right range - for example in the Graphic equalizer, the sliders in the normal gui have ranges -12 to 12, but the equivalent spin control in the plain gui has range 0 to 1.
2. A wxSpinCtrlDouble can be used for a property which has discrete values. In the normal view of the tube amplifier, there's a mode which can be set to be mode 1 or mode 2. In the plain gui, there's a wxSpinCtrlDouble for the mode, with a range 0 to 1.0

Checked the VST3 versions in Reaper, and the plain gui version, which is just a series of sliders and values, does not have the first isssue above - the values are in the correct range.

There are also accessibility isssues, but it probably best to address them after above problems have been sorted out.

_Originally posted by @DavidBailes in https://github.com/audacity/audacity/issues/2295#issuecomment-1038980633_

**Comment by DavidBailes** [comment]
Here's another vst3 plugin, for which the plain gui has serious problems (ie it's unusable):
https://www.kvraudio.com/product/valhallafreqecho-by-valhalla-dsp
I installed both the vst and vst3 versions:

Audacity's normal gui for the vst3 version:
![image](https://user-images.githubusercontent.com/11887021/154081648-bf6f3594-b8db-49ac-b9cd-5c1c2c7e3f1a.png)

Audacity's plain gui for the vst3 version:
![image](https://user-images.githubusercontent.com/11887021/154081893-a07f5c12-938b-4e7a-99b3-3e11c8c81e74.png)

Audacity's plain gui for the vst version:
![image](https://user-images.githubusercontent.com/11887021/154082093-f936e90a-cba6-47d4-aa2a-289308e05e85.png)


Reaper's plain gui of the vst3 version:
![image](https://user-images.githubusercontent.com/11887021/154081159-4a06a03c-69bf-4f7d-9c6b-a16d5f64b330.png)

**Comment by DavidBailes** [comment]
Values also aren't in the right range for two VST3 plugins by Tokyo Dawn: Kotelnikov and VOS SlickEQ

**Comment by dozzzzer** [comment]
Please see the following example. 
Here's the UI version of the free VST3 Izotope Vinyl plugin:
 
![image](https://user-images.githubusercontent.com/70063970/172157705-b37342a8-8228-4ac1-ac6c-d7cfbb0d9433.png)

Here's a short demonstration of the controls not changing the values properly (only the default values are shown regardless of the slider's position):

https://user-images.githubusercontent.com/70063970/172157794-c7fe189d-ee93-4fcf-b415-5f4af428080c.mp4

**Comment by dozzzzer** [comment]
Exact same issue was observed with Kickstart 2 plugin:
https://www.kickstart-plugin.com/
